### ISSUE TITLE
VFS4G verbs should fail if KernelDriver.RegisterForOfflineIO() fails

### ISSUE THREAD

**Issue by wilbaker** [microsoft/VFSForGit#1474]
Currently VFS4G is not checking the return value of `RegisterForOfflineIO()`.  If `RegisterForOfflineIO()` fails the verb should not be allowed to continue as its attempts to perform offline IO will likely fail.

Steps to reproduce `RegisterForOfflineIO` failing:

1. Install an official VFS4G on Mac
2. Build a debug build of VFS4G locally
3. Attempt to clone a repo

*Output*

```
~/Repos/VFSForGit/src>./Scripts/Mac/GVFS_Clone.sh https://gvfs.visualstudio.com/ci/_git/ForTests
PrjFS kernel service interface version mismatch. Kernel: 0.6.19224.2, this library expects: 0.2.173.2
```

**Comment by wilbaker** [comment]
@jrbriggs @jeschu1 I just came across this problem and filed this issue.  For now I've made it pri2, if you think it should be pri1 or pri1/pri2 please feel free to update the labels.

**Comment by jeschu1** [comment]
@wilbaker, I agree with the prioritization here.  The change is simple however the only time we are currently concerned with enabling offlineIO is functional tests.

**Comment by wilbaker** [comment]
> the only time we are currently concerned with enabling offlineIO is functional tests.

@jeschu1 the functional tests are only one reason we need this method.  The `gvfs` verbs register for offline IO because some of them need to perform offline I/O.  See comment in #1429 for an example:

> During gvfs clone, the git subprocess itself wants to create files/directories in the offline root, so we need it to inherit the parent gvfs process's entitlement to access offline roots. The exception check therefore now walks up the process hierarchy.

This means that if `RegisterForOfflineIO` fails the verb might fail due to having its I/O blocked, and it might not be clear why.  

At the time `RegisterForOfflineIO` I don't think we have a tracer available and so to the user (and whoever is on support) it would appear that the verb's I/O mysteriously failed.

This is an easy/quick change, and so I think we should pick it up as soon as the pri1s are complete (or one of us can grab it in parallel if we find ourselves with time to pick it up).

**Comment by jeschu1** [comment]
@wilbaker, with that in mind I'll pick it up in parallel.  It's a such an easy change might as well!
